src/Components/Modals/ActivationKeyWizard.js

```diff
@@ -206,40 +236,86 @@ const CreateActivationKeyWizard = ({
-                  },
-                  onError: () => {
-                    addErrorNotification('Something went wrong', {
-                      description:
-                        'Your changes could not be saved. Please try again.',
+            if (step.id === 4) {
+              const mutationFn = isEditMode
+                ? updateActivationKey
+                : createActivationKey;
+              const mutationData = isEditMode
+                ? {
+                    activationKeyName: activationKey.name,
+                    description,
+                    role,
+                    serviceLevel: sla,
+                    usage,
+                    releaseVersion: extendedReleaseVersion,
+                  }
+                : {
+                    name,
+                    description,
+                    role,
+                    serviceLevel: sla,
+                    usage,
+                    additionalRepositories: workload.includes('Extended')
+                      ? extendedReleaseRepositories
+                      : undefined,
+                    releaseVersion: extendedReleaseVersion,
+                  };
+              mutationFn(mutationData, {
+                onSuccess: (updatedData) => {
+                  if (isEditMode) {
+                    queryClient.invalidateQueries([
+                      `activation_key_${activationKey.name}`,
+                    ]);
+                    queryClient.setQueryData([`activation_key`], (prev) => {
+                      if (!prev) return prev;
+                      return prev.map((key) =>
+                        key.activationKeyName === activationKey.name
+                          ? { ...key, description: updatedData.description }
+                          : key
+                      );
                     });
-                    onClose();
-                  },
-                }
-              );
+                    setDescription(updatedData.description);
+                    queryClient.resetQueries([
+                      `activation_key_${activationKey.name}`,
+                    ]);
+                    addSuccessNotification(
+                      `Changes saved for activation key "${activationKey.name}"`
+                    );
+                  } else {
+                    addSuccessNotification(`Activation key "${name}" created`);
+                  }
+                },
+                onError: () => {
+                  addErrorNotification(
+                    isEditMode
+                      ? `Error updating activation key ${activationKey.name}.`
+                      : 'Something went wrong',
+                    isEditMode
+                      ? undefined
+                      : {
+                          description:
+                            'Your changes could not be saved. Please try again.',
+                        }
+                  );
+                  onClose();
+                },
+              });
```

Comment:
Suggestion: It might be overkill but for simplicities sake, I would just change this to invalidate the current activation key and the list of activation key queries so that you can avoid manually manipulating query data (via `queryClient.setQueryData()`)